### ISSUE TITLE
commit array should be present in change events

### ISSUE THREAD

**Issue by xibz** [cdevents/spec#150]
Currently, the change events include no notion or mention of commit(s). The only time where it would look to be appropriate is to have an `id` that is the commit. However, that seems odd imo, as things could be multiple commits. You could then say, make it the feature branch, but then you'd need to look up the commits from the feature branch to get that metadata. So I think this shows a little bit of overloading of what an `id` is.

So this proposal is to add a `commits` array
```bash
"commits": [
  {
    "id": <hash>,
    "author": "xibz" # or email?
  }
]
```

This makes it really clear on what was merged and by who.

**Comment by afrittoli** [comment]
Thanks @xibz - indeed until now we've worked with a few simplifications in place, like only one ID in changes and only one artifact in a service/deployment. This needs to be solved by either expecting individual events for each part or by extending the existing data model to include lists of subjects.

Specifically for commits, the minimal approach would be to include the `id` only.

**Comment by xibz** [comment]
Yep! So what is the action item here? I can add that to the changed events, or were we thinking of a design doc or something to see how this works? Im for either. Maybe I can setup a meeting and we can do a quick go over on a few ideas on how to solve this.

**Comment by e-backmark-ericsson** [comment]
The content of the source change events has not been much elaborated yet. Before going into details we'd need to settle on whether a source change event should describe an individual commit or if it should describe a full pull/merge request. So far I believe our discussions have assumed that one source change in CDEvents lingo represents a full pull request, but there could be an option to instead let a source change correspond to an individual commit instead, and then have another event representing a pull request.

**Comment by mekhanique** [comment]
Piggy backing on this.  There is a use case where change sets could be related to or dependent upon change sets in other repositories.  It is foreseable that some form of merge management could execute upon related change sets.  I'd suggest an object that can support a larger change set combination to support this.  

Also, it might be beneficial to capture commit signatures for auditing purposes.  Here is an example data structure:
